### PR TITLE
Remove `PRIMER3HOME` in favor of package-relative paths

### PR DESCRIPTION
**Pull request by benpruitt**
Includes incidental fix in `thermoanalysis.pyx`
- `c_ascii_structure[0] = '\0';` -> `c_ascii_structure[0] = b'\0';`

### CODE REVIEW

File: primer3/wrappers.py [hunk 1]
```diff
@@ -501,7 +494,7 @@ def designPrimers(
     )
     p3_args.setdefault(
         'PRIMER_THERMODYNAMIC_PARAMETERS_PATH',
-        pjoin(LIBPRIMER3_PATH, 'primer3_config/'),
+        pjoin(LIBPRIMER3_PATH, 'primer3_config'),
     )
```
Comment (grinner) [inline primer3/wrappers.py:498]: Should this just be `THERMO_PATH`?

**Review by grinner — COMMENTED** [review]
One question

File: primer3/wrappers.py (hunk 1 above)
Comment (benpruitt) [reply primer3/wrappers.py:498]: Oh good call -- fixed

**Review by grinner — APPROVED** [review]
lgtm

**Pull request merged**